packages/js/components/src/sortable/sortable.tsx

```diff
@@ -67,43 +95,140 @@ export const Sortable = ( {
 		event: DragEvent< HTMLDivElement >,
 		index: number
 	) => {
-		if (
-			dropIndex !== null &&
-			dragIndex !== null &&
-			dropIndex !== dragIndex
-		) {
-			const nextItems = moveIndex( dragIndex, dropIndex, items );
-			setItems( nextItems as JSX.Element[] );
-			onOrderChange( nextItems );
-		}
-
-		setTimeout( () => {
-			setDragIndex( null );
-			setDropIndex( null );
-			onDragEnd( event );
-		}, THROTTLE_TIME );
+		persistItemOrder();
+		onDragEnd( event );
 	};
 
 	const handleDragOver = (
 		event: DragEvent< HTMLLIElement >,
 		index: number
 	) => {
-		const targetIndex = isBefore( event, isHorizontal ) ? index : index + 1;
+		if ( dragIndex === null ) {
+			return;
+		}
+
+		// Items before the current item cause a one off error when
+		// removed from the old array and spliced into the new array.
+		let targetIndex = dragIndex < index ? index : index + 1;
+		if ( isBefore( event, isHorizontal ) ) {
+			targetIndex--;
+		}
+
 		setDropIndex( targetIndex );
 		onDragOver( event );
 	};
 
 	const throttledHandleDragOver = useCallback(
 		throttle( handleDragOver, THROTTLE_TIME ),
-		[]
+		[ dragIndex ]
 	);
 
+	const handleKeyDown = (
+		event: KeyboardEvent< HTMLLIElement >,
+		index: number
+	) => {
+		const { key } = event;
+		const isSelecting = dragIndex === null || dropIndex === null;
+		const selectedLabel = getItemName( ref.current, selectedIndex );
+
+		if ( key === ' ' ) {
```

Comment:
Optionally might be worth a comment here mentioning that a key value of `" "` is the spacebar. Odd that there isn't a more explicit key name.